### PR TITLE
Null Check added to GetFullPath Function in Unix

### PR DESCRIPTION
**Pull request by Anipik**

### CODE REVIEW

File: src/mscorlib/shared/System/IO/Path.Unix.cs [hunk 1]
```diff
@@ -23,6 +23,8 @@ public static string GetFullPath(string path)
             if (path.Length == 0)
                 throw new ArgumentException(SR.Arg_PathEmpty, nameof(path));
 
+            PathInternal.CheckInvalidPathChars(path);
```
Comment (JeremyKuhne) [inline src/mscorlib/shared/System/IO/Path.Unix.cs]: You should just do the inline check here just like Windows and take the opportunity to clean out the unused code. HasIllegalCharacters() isn't used anywhere anymore, and we don't want people to slip into using it.

File: src/mscorlib/shared/System/IO/PathInternal.cs [hunk 2]
```diff
@@ -9,7 +9,7 @@ namespace System.IO
 {
     /// <summary>Contains internal path helpers that are shared between many projects.</summary>
     internal static partial class PathInternal
-    {
+    {
```
Comment (JeremyKuhne) [inline src/mscorlib/shared/System/IO/PathInternal.cs]: Added some spaces here.

**Comment by Anipik** [comment]
@JeremyKuhne  it doesnot look like being caused by this change . Can you confirm that ?. The build is successful at my local machine
12:09:47 Tools can be found at http://llvm.org/releases/download.html#3.8.0
12:09:47 Downloading formatting tools
12:09:52 Invoke-WebRequest : Could not find a part of the path 
12:09:52 'D:\j\workspace\x64_windows_n---616f30f2\jitutils\bin\clang-format.exe'.
12:09:52 At line:1 char:1
12:09:52 + Invoke-WebRequest -Uri 
12:09:52 https://clrjit.blob.core.windows.net/clang-tools/windows/ ...
12:09:52 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
12:09:52 ~~~
12:09:52     + CategoryInfo          : NotSpecified: (:) [Invoke-WebRequest], Directory 
12:09:52    NotFoundException
12:09:52     + FullyQualifiedErrorId : System.IO.DirectoryNotFoundException,Microsoft.P 
12:09:52    owerShell.Commands.InvokeWebRequestCommand
12:09:52  
12:10:08 Invoke-WebRequest : Could not find a part of the path 
12:10:08 'D:\j\workspace\x64_windows_n---616f30f2\jitutils\bin\clang-tidy.exe'.
12:10:08 At line:1 char:1
12:10:08 + Invoke-WebRequest -Uri 
12:10:08 https://clrjit.blob.core.windows.net/clang-tools/windows/ ...
12:10:08 + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
12:10:08 ~~~
12:10:08     + CategoryInfo          : NotSpecified: (:) [Invoke-WebRequest], Directory 
12:10:08    NotFoundException
12:10:08     + FullyQualifiedErrorId : System.IO.DirectoryNotFoundException,Microsoft.P 
12:10:08    owerShell.Commands.InvokeWebRequestCommand
12:10:08  
12:10:08 Jitutils not built!
12:10:08 
12:10:08 D:\j\workspace\x64_windows_n---616f30f2>exit -1

**Comment by Anipik** [comment]
@dotnet-bot test Windows_NT x64 Innerloop Formatting

**Comment by Anipik** [comment]
@dotnet-bot   Windows_NT x86 Checked Innerloop Build and Test
rikicking as there was a failure due to lack of space

**Comment by Anipik** [comment]
@dotnet-bot test Windows_NT x86 Checked Innerloop Build and Test

**Comment by Anipik** [comment]
Rekicking these  legs due to lack of space failure

**Comment by Anipik** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by Anipik** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by jkotas** [comment]
@safern The CoreLib mirror is not picking up this PR. Could you please take a look?

**Comment by safern** [comment]
Sorry I was OOF today. Will take a look.

**Pull request merged**